Shared/Samples/Augment reality to navigate route/AugmentRealityToNavigateRouteView.swift

```diff
@@ -14,245 +14,239 @@
-                        routeResult: routeResult,
-                        routeGraphic: routeGraphic
-                    )
-                }
-                .task {
-                    try? await elevationSource.load()
-                }
-        } else {
-            VStack(spacing: 0) {
-                WorldScaleSceneView { _ in
-                    SceneView(scene: scene, graphicsOverlays: [graphicsOverlay])
-                }
-                .calibrationButtonAlignment(.bottomLeading)
-                .onCalibratingChanged { isPresented in
-                    scene.baseSurface.opacity = isPresented ? 0.6 : 0
-                }
-                .task {
-                    try? await locationDataSource.start()
-                    
-                    for await location in locationDataSource.locations {
-                        try? await routeDataModel.routeTracker?.track(location)
-                    }
-                }
-                .overlay(alignment: .top) {
-                    Text(statusText)
-                        .multilineTextAlignment(.center)
-                        .frame(maxWidth: .infinity, alignment: .center)
-                        .padding(8)
-                        .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
-                }
-                .onDisappear {
-                    Task { await locationDataSource.stop() }
-                }
-                Divider()
+        MapView(
+            map: model.map,
+            graphicsOverlays: model.graphicsOverlays
+        )
+        .locationDisplay(model.locationDisplay)
+        .onSingleTapGesture { _, mapPoint in
+            tapLocation = mapPoint
+        }
+        .task(id: tapLocation) {
+            guard let tapLocation else { return }
+            
+            do {
+                try await model.addRouteGraphic(for: tapLocation)
+            } catch {
+                self.error = error
             }
-            .toolbar {
-                ToolbarItemGroup(placement: .bottomBar) {
-                    Button("Start") {
-                        isNavigating = true
-                        Task {
-                            do {
-                                try await startNavigation()
-                            } catch {
-                                print("Failed to start navigation.")
-                            }
-                        }
-                    }
-                    .disabled(isNavigating)
-                }
+        }
+        .task {
+            do {
+                try await model.setUp()
+            } catch {
+                self.error = error
+            }
+        }
+        .overlay(alignment: .top) {
+            instructionText
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                toolbarButtons
             }
         }
+        .onAppear {
+            model.statusText = "Tap to place a start point."
+        }
```

Comment:
This could just be the default string for `statusText` rather than assigning it when the view appears, right? Not a big deal though.